rotkehlchen/exchanges/iconomi.py

```diff
@@ -0,0 +1,260 @@
+            raise RemoteError('ICONOMI returned invalid non-dict response')
+
+        return json_ret
+
+    def validate_api_key(self) -> Tuple[bool, str]:
+        "Validates that the ICONOMI API key is good for usage in Rotki"
+
+        try:
+            self._api_query('get', 'user/balance')
+            return True, ""
+
+        except RemoteError:
+            return False, 'Provided API Key is invalid'
+
+    def query_balances(self, **kwargs: Any) -> Tuple[Optional[Dict[Asset, Dict[str, Any]]], str]:
+        balances = {}
+        resp_info = self._api_query('get', 'user/balance')
+
+        for balance_info in resp_info['assetList']:
+            ticker = balance_info['ticker']
+            try:
+                asset = iconomi_asset(ticker)
+                balances[asset] = {
+                    'amount': balance_info['balance'],
+                    'usd_value': balance_info['value'],
+                }
+            except UnknownAsset:
+                self.msg_aggregator.add_warning(
+                    f'Found unsupported ICONOMI asset {ticker}. '
+                    f' Ignoring its balance query.',
+                )
+
+        for balance_info in resp_info['daaList']:
+            ticker = balance_info['ticker']
+            try:
+                if ticker not in SUPPORTED_FUND_TICKERS:
+                    raise UnknownAsset(ticker)
+
+                asset = iconomi_asset(ticker)
+                balances[asset] = {
+                    'amount': balance_info['balance'],
+                    'usd_value': balance_info['value'],
+                }
+            except UnknownAsset:
+                self.msg_aggregator.add_warning(
+                    f'Found unsupported ICONOMI strategy {ticker}. '
+                    f' Ignoring its balance query.',
+                )
+
+        return (balances, "")
+
+    def query_online_trade_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> List:
+
+        page = 0
+        all_transactions = []
+
+        while True:
+            resp = self._api_query('get', 'user/activity', {"pageNumber": str(page)})
+
+            if len(resp['transactions']) == 0:
+                break
+
+            all_transactions.extend(resp['transactions'])
+            page += 1
+
+        log.debug('ICONOMI trade history query', results_num=len(all_transactions))
+
+        trades = []
+        for tx in all_transactions:
+            timestamp = tx['timestamp']
+            if timestamp and timestamp < start_ts:
+                continue
+            if timestamp and timestamp > end_ts:
+                continue
+
+            if tx['type'] in ('buy_asset', 'sell_asset', 'buy_fund', 'sell_fund'):
+                try:
+                    trades.append(trade_from_iconomi(tx))
+                except UnknownAsset:
+                    log.warning('Ignoring transaction %s because of unsupported asset', str(tx))
```

Comment:
Please use f-string notation (for consistency's sake with the rest of the codebase)